src/fastoad/cmd/doe.py

```diff
@@ -0,0 +1,376 @@
+"""Python module containing the tools for generating and postprocessing Design of
+Experiments (DOEs) in FAST-OAD."""
+#  This file is part of FAST-OAD : A framework for rapid Overall Aircraft Design
+#  Copyright (C) 2025 ONERA & ISAE-SUPAERO
+#  FAST is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation, either version 3 of the License, or
+#  (at your option) any later version.
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#  You should have received a copy of the GNU General Public License
+#  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import itertools
+import warnings
+from dataclasses import dataclass, field
+from os import PathLike
+from typing import ClassVar, Dict, List, Optional, Union
+
+import numpy as np
+import pandas as pd
+import smt.sampling_methods as samp
+from smt.applications.mfk import NestedLHS
+from smt.sampling_methods.sampling_method import ScaledSamplingMethod
+
+import fastoad.api as oad
+from fastoad._utils.files import as_path
+from fastoad.openmdao.variables import VariableList
+
+
+@dataclass
+class DOEVariable:
+    """Represents a Design of Experiments (DOE) variable defined by its unique `id`.
```

Comment:
Same overly zealous remark as for the module docstring: I prefered if the start of the docstring is not on the same line as the `"""` if you have multiple lines ^^